src/bin/drivers/xbee/xbee.cpp

```diff
@@ -345,14 +342,14 @@ void XBeeDevice::process_frame_at_command_response(const string& response_string
 
     if (at_command == "SH") {
         assert(response->command_status == 0);
-        uint32_t upper_serial_number = *((uint32_t *)&response->command_data_start);
-        my_serial_number |= (SerialNumber)upper_serial_number;
+        uint32_t upper_serial_number = big_to_native(*((uint32_t*)&response->command_data_start));
+        my_serial_number |= ((SerialNumber)upper_serial_number << 32);
     }
 
     if (at_command == "SL") {
         assert(response->command_status == 0);
-        uint32_t lower_serial_number = *((uint32_t *)&response->command_data_start);
-        my_serial_number |= ((SerialNumber) lower_serial_number << 32);
+        uint32_t lower_serial_number = big_to_native(*((uint32_t*)&response->command_data_start));
+        my_serial_number |= ((SerialNumber)lower_serial_number);
         glog.is_verbose() && glog << "serial_number= " << std::hex << my_serial_number << std::dec << " node_id= " << my_node_id  << " (this device)" << endl;
 
         // Broadcast our node_id, and request everyone else's node_id
```

Comment:
I know that we ignore this broadcast, but is there a need to do it in the first place?